pkg/services/ngalert/models/alert_rule.go

```diff
@@ -423,6 +424,42 @@ func (alertRule *AlertRule) PreSave(timeNow func() time.Time) error {
 	return nil
 }
 
+// ValidateAlertRule validates various alert rule fields.
+func (alertRule *AlertRule) ValidateAlertRule(cfg setting.UnifiedAlertingSettings) error {
```

Comment:
Personally, I feel that function would be better. Or even a validator struct because the correct domain model validation (structural is done in API layer) requires folder service and datasource service, and expr engine.